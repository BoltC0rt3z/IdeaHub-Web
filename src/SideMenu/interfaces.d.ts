export interface Props extends MenuProps {
  collapsed: boolean | undefined;
  openDrawer: boolean;
  closeDrawer: () => void;
}
