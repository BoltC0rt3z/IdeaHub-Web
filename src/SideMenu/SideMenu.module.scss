@import '../styles/styles.module.scss';

.sider {
    :global(.ant-layout-sider-children) {
        background-color: $color-dark-blue-lighter;
    }
}