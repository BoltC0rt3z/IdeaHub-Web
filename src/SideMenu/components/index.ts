import MenuTitle from "./MenuTitle";
import Menu from "./Menu";

export { MenuTitle, Menu };
