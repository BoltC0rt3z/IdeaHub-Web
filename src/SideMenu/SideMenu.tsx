import React from "react";
import { Layout, Drawer } from "antd";
import { MenuProps } from "antd/lib/menu";
import { useMedia } from "react-use";

import { Menu } from "./components";
import { DeviceMeasures } from "../utils";

import styles from './SideMenu.module.scss';

interface Props extends MenuProps {
  collapsed: boolean | undefined;
  openDrawer: boolean;
  closeDrawer: () => void;
}

const { Sider } = Layout;

const SideMenu: React.FC<Props> = props => {
  const { collapsed, openDrawer, closeDrawer } = props;
  const isMobile: boolean = useMedia(DeviceMeasures.MOBILE);

  return !isMobile ? (
    <Sider collapsible collapsed={collapsed} trigger={null} collapsedWidth={0} className={styles.sider}>
      <Menu />
    </Sider>
  ) : (
      <Drawer
        placement="left"
        closable={false}
        visible={openDrawer}
        getContainer={false}
        bodyStyle={{ padding: 0 }}
        onClose={closeDrawer}
      >
        <Menu />
      </Drawer>
    );
};

export default SideMenu;
