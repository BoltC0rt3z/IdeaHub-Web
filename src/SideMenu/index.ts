import SideMenu from "./SideMenu";
import { MenuTitle } from "./components";

export { SideMenu, MenuTitle };
