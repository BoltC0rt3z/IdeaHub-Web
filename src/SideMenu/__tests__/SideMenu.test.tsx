import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import SideMenu from '../SideMenu';
import { Props } from '../interfaces';

describe('Tests for the sideMenu', () => {
  const props: Props = {
    collapsed: false,
    openDrawer: true,
    closeDrawer: jest.fn(),
  };

  const wrapper: ReactWrapper = mount(
    <BrowserRouter>
      <SideMenu {...props} />
    </BrowserRouter>,
  );

  it('should render the SideMenu without crushing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
