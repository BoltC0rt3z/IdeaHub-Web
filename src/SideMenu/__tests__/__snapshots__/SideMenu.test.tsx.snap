// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the sideMenu should render the SideMenu without crushing 1`] = `ReactWrapper {}`;
